lib/iso_spinner.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (C) IBM Corp. 2016.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import glob
+import logging
+import os
+
+from lib import exception
+from lib import utils
+from lib import distro_utils
+
+LOG = logging.getLogger(__name__)
+
+
+class MockPungiSpinner(object):
+
+    def __init__(self, config):
+        self.config = config
+        (self.distro, self.version, self.arch) = distro_utils.detect_distribution()
+        # only the major number must be considered
+        self.version = self.version.split('.')[0]
+
+    def _run_mock_command(self, cmd):
+        try:
+            utils.run_command("mock -r %s %s" % (self.config.get('mock_config'),
+                                                 cmd))
+        except exception.SubprocessError:
+            LOG.error("Failed to spin ISO")
+            raise
+
+    def build(self):
+        LOG.info("Starting ISO spin process")
+        self._setup()
+        self._spin()
+        self._save()
+
+    def _setup(self):
+        LOG.info("Initializing a chroot")
+        self._run_mock_command("--init")
+
+        package_list = ["createrepo", "pungi"]
+        LOG.info("Installing %s inside the chroot" % " ".join(package_list))
+        self._run_mock_command("--install %s" % " ".join(package_list))
+
+        self._create_spin_repo()
+
+        self._create_spin_kickstart()
+
+    def _create_spin_repo(self):
+        LOG.info("Creating spin repository inside chroot")
+
+        LOG.debug("Creating spin repo directory")
+        mock_spin_repo_dir = self.config.get('mock_spin_repo').get('dir')
+        self._run_mock_command("--shell 'mkdir %s'" % mock_spin_repo_dir)
+
+        LOG.debug("Copying rpm to spin repo directory")
+        rpm_dir = self.config.get('rpm_dir')
+        rpm_files = glob.glob(os.path.join(rpm_dir, "*.rpm"))
+        self._run_mock_command("--copyin %s %s" %
+                               (" ".join(rpm_files), mock_spin_repo_dir))
+
+        LOG.debug("Creating comps.xml")
+        # TODO: create comps.xml
+        comps_xml_file = "hostos-comps.xml"
+        comps_xml_chroot_path = "/"
+        self._run_mock_command("--copyin %s %s" %
+                               (comps_xml_file, comps_xml_chroot_path))
+
+        LOG.debug("Creating spin repo")
+        createrepo_cmd = "createrepo -v -g /%s %s" % (comps_xml_file,
```

Comment:
So we could use `comps_xml_chroot_path` here and make the link with the _run_mock_command parameters above more obvious to people unfamiliar with mock.